pkg/util/validation/validate.go

```diff
@@ -33,7 +33,7 @@ const (
 	errInvalidLabel       = "sample invalid label: %.200q metric %.200q"
 	errLabelNameTooLong   = "label name too long: %.200q metric %.200q"
 	errLabelValueTooLong  = "label value too long: %.200q metric %.200q"
-	errTooManyLabels      = "sample for '%s' has %d label names; limit %d"
+	errTooManyLabels      = "series has too many label names (actual: %d, limit: %d) series: '%s'"
```

Comment:
Nit: We can save few characters and show more from the series.

```suggestion
	errTooManyLabels      = "series has too many labels (actual: %d, limit: %d): '%s'"
```